### ISSUE TITLE
Fails with lines-tables-only

### ISSUE THREAD

**Issue by e00E** [pacak/cargo-show-asm#187]
This option is documented [here](https://doc.rust-lang.org/cargo/reference/profiles.html#debug).

```
./src/lib.rs:
pub fn foo() {}

./Cargo.toml:
[package]
name = "a"
version = "0.1.0"
edition = "2021"

[dependencies]

[profile.release]
debug = "line-tables-only"
```

`cargo asm` fails with "Error: No artifact found" when this option is set. It works when it is set to `0`, `1`, `2`.

**Comment by pacak** [comment]
Managed to reproduce it, but had to update to 1.71

So I'm asking rustc to produce `.s` files and json output, then I'm parsing that json with cargo-metadata to find out what was generated. json contains this bit of info which supposed to parse into `ArtifactProfile`

```json
"profile":{"opt_level":"3","debuginfo":"line-tables-only","debug_assertions":false,"overflow_checks":false,"test":false},
```

https://docs.rs/cargo_metadata/latest/cargo_metadata/struct.ArtifactProfile.html

And `debuginfo` field is described as `Option<u32>`...

I think once `cargo-metadata` gets updated - `cargo-show-asm` will magically start working again...

**Comment by pacak** [comment]
Made a ticket, we'll see what they reply.

**Comment by pacak** [comment]
https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Breaking.20JSON.20change.20in.201.2E71

**Comment by pacak** [comment]
I released 0.2.20 with a workaround for this problem. Will close the ticket with a proper fix once cargo-metadata updates.
